### ISSUE TITLE
[DOCS] improve airgapped customers missing documentation for security artifacts

### ISSUE THREAD

**Issue by 111andre111** [elastic/security-docs#2541]
## Description 

At the moment there is only a documentation for airgapped customers how to deal with the beats binary artifacts here for `artifacts.elastic.co`:
https://www.elastic.co/guide/en/fleet/8.4/air-gapped.html#host-artifact-registry
as well as for getting integrations from a local epr for `epr.elastic.co`:
https://www.elastic.co/guide/en/fleet/8.4/air-gapped.html#air-gapped-diy-epr
But the security artifacts for endpoint security for an airgapped customer are missing:
The endpoints need contact to follow 2 additional domains:
`artifacts.security.elastic.co`
`cloud.security.elastic.co`

## Acceptance Test Criteria

Make clear how the contact to domains:
`artifacts.security.elastic.co`
`cloud.security.elastic.co`
works and how to deal with the same for airgapped customers 

clarify advanced settings for the integration

clarify the contact to `cloud.security.elastic.co`

## Notes

It would be good if we could clarify the following:

1. In security integration there exist several settings:
windows.advanced.artifacts.global.base_url, linux.advanced.artifacts.global.base_url, mac.advanced.artifacts.global.base_url
windows.advanced.artifacts.global.manifest_relative_url, linux.advanced.artifacts.global.manifest_relative_url, mac.advanced.artifacts.global.manifest_relative_url

As an example for linux looks like that:
linux.advanced.artifacts.global.base_url                 default https://artifacts.security.elastic.co/
linux.advanced.artifacts.global.manifest_relative_url    default /downloads/endpoint/manifest/artifacts-<version>.zip

So for downloading a manifest for version 8.4.0 the complete url lookup looks like that:
`https://artifacts.security.elastic.co/downloads/endpoint/manifest/artifacts-8.4.0.zip`
Inside the manifest are a sig file and a json file.

Looking to the json file it has again a relative_url field which goes back to the base_url config field as reference:
```
# cat manifest.json | grep  relative_url | awk '{print "https://artifacts.security.elastic.co "$2}' | sed 's/ "//g' | sed 's/"$//g'
https://artifacts.security.elastic.co/downloads/endpoint/diagnostic-configuration-v1/11b4c72e47d4b2651915cb741305e775b3a54bf0a247e0acc3021bdc49d0e636
https://artifacts.security.elastic.co/downloads/endpoint/diagnostic-endpointelf-v1-blocklist/5093bb6536c76a5c6d90c2588e552a74845710fe8af4b8adb2967320dd771a19
https://artifacts.security.elastic.co/downloads/endpoint/diagnostic-endpointelf-v1-exceptionlist/30aeab4ad131ea70bd21e9c2e9528b4d178279aa3c8fe6de57c5645e49a185a9
https://artifacts.security.elastic.co/downloads/endpoint/diagnostic-endpointelf-v1-model/98611a30e1ce18f374d6e3d81934581f475cf644cf56434120915de660ca6a6b
https://artifacts.security.elastic.co/downloads/endpoint/diagnostic-endpointmacho-v1-blocklist/e2637d1c5dd39719423ce67b7ca8ad3ca1286b508f1336e5479f85b5a9abb228
...
...
```
So if the `base_url` setting would be changed to another main server like e.g. `localserver`, the manifest lookup would change to `https://localserver/downloads/endpoint/manifest/artifacts-8.4.0.zip` and the follow-up artifacts to e.g. `https://localserver/downloads/endpoint/diagnostic-configuration-v1/11b4c72e47d4b2651915cb741305e775b3a54bf0a247e0acc3021bdc49d0e636`

That way the whole endpoint infrastructure could ask an internal server.
However it's still the duty of the customer to download the artifacts from internet to the local server similar to point 1.a outlined in this artifacts documentation:
https://www.elastic.co/guide/en/fleet/8.4/air-gapped.html#host-artifact-registry
And it's important to point out that the endpoint security binary on the endpoint initiates this connection individually, not kibana.

2. It might be important to point out, that all manifests get actualized under the hood with a newer manifest version possibly:
```
# cat manifest.json | jq | grep manifest_version
    "manifest_version": "1.0.385",
```
and we update if a newer manifest version comes out all version artifacts to a newer manifest version.
So my conclusion would be customers would need to do a script or something like that to investigate if a new manifest version came out and new artifacts would be needed to be downloaded.

3. Additionally it is unfortunately missing and in depth description of what all the advanced settings in the endpoint security integration are for.
https://docs.elastic.co/integrations/endpoint
There are some hinting blobs that give some guidance, but don't help for an in depth insight and it would be good if we set a more clear documentation here:
![Screenshot 2022-10-05 copy](https://user-images.githubusercontent.com/48132449/194006231-9c5efd38-68f6-4097-8a49-95417907cdd0.png)

4. Finally in case an alert gets raised the endpoint initiates a connection to this domain as a first sha lookup:
https://cloud.security.elastic.co/v1/lookup/<sha-value>
And it's important to point out again for this one, that the endpoint security binary on the endpoint initiates this connection individually, not kibana.
I not even sure here, if there is any good way how to replicate this one for airgapped customers.
It could be clarified that the lookup happens only on Windows and MacOS endpoints as far as I can see:
https://github.com/elastic/endpoint-dev/blob/main/Plugins/Policy/Iface/plugins/Policy_Alerts.h#L21

@elastic/threat-data-services  @jmikell821

**Comment by 111andre111** [comment]
Seems like this is ongoing here:
https://github.com/elastic/security-docs/pull/2791

**Comment by 111andre111** [comment]
`cloud.security.elastic.co`
I think this is needed for feature `Reputation service`

**Comment by intxgo** [comment]
there's already a dedicated doc for airgapped cases https://www.elastic.co/guide/en/security/current/offline-endpoint.html

**Comment by 111andre111** [comment]
Yes, but we don't nowhere outline that this endpoint mentioned is used for Reputation Service.

So partially it's still not there what has been outlined here regarding the endpoint.

**Comment by intxgo** [comment]
> Yes, but we don't nowhere outline that this endpoint mentioned is used for Reputation Service.
> 
> So partially it's still not there what has been outlined here regarding the endpoint.

Quite the contrary, reputation services are available only in cloud subscriptions. Endpoint won't ever try to reach those URL endpoints in non-cloud deployment. 

This page already clarifies the availability of the feature https://www.elastic.co/guide/en/security/current/configure-endpoint-integration-policy.html
```
Reputation service requires an active [Platinum or Enterprise subscription](https://www.elastic.co/pricing) and is available on cloud deployments only.
```

**Comment by fdartayre** [comment]
+1 @111andre111, It would be good to have the domains all documented in the same page

**Comment by 111andre111** [comment]
Agreed it can be on that page too, but most importantly it should be finally documented...

**Comment by intxgo** [comment]
We already provide guide for air gapped environments, which I think was sufficient for the customers
https://www.elastic.co/guide/en/security/current/offline-endpoint.html

Here's a bit information you've asked for:

1. https://artifacts.security.elastic.co

Security artifacts CDN (Content Delivery Network). Endpoint is fetching security artifacts updates periodically.

2. https://cloud.security.elastic.co/v1/manifest

New cloud API orchestrating staged artifacts rollout. It controls which artifacts Endpoint should use/download. The actual CDN URL of the artifacts manifest is provided in response, rather than using the fixed one described in point 1.

Precondition:
Several conditions have to be met to make Endpoint eligible for candidate artifacts testing. Roughly it must be a default cloud deployment, healthy, with telemetry, without customizations by advanced policy. Endpoints not eligible for testing (for example air gapped) don't even call it, but use the legacy Security artifacts CDN directly.

3. https://cloud.security.elastic.co/v1/lookup

Quick lookup for a hash of detected malware. Aiming to rapidly reduce known false positives, before such Endpoint fetches full update of security artifacts.

Controlled by advanced policy:
`advanced.alerts.cloud_lookup` boolean switch
`advanced.alerts.cloud_lookup_url` url override

4. https://cloud.security.elastic.co/v1/alerts-lookup/

Another service for reducing false positives but also to improve our detection. Endpoint sends the actual alert but redacted, carefully stripped from PII for privacy, to the service instead of only the hash. This helps us to learn immediately about new malware campaigns, and collect an essential sample from such endpoint. When the service decides it's worth to collect the sample, it provides a preauthenticated URL for the upload in its response.

Controlled by advanced policy:
`advanced.alerts.sample_collection` boolean switch
`advanced.alerts.reputation_url` url override
`advanced.alerts.reputation_auth` authentication override

5. https://telemetry.elastic.co/v3/send/endpoint-health

Telemetry closely related with the new security artifacts API, listed in point 2. Endpoint sends telemetry about health before checking the API. If instructed to test candidate artifacts, Endpoint additionally sends telemetry documents about start and abort reason of the candidate artifacts testing.

Precondition:
Telemetry in the stack must be enabled, cloud deployment (not air gapped), etc.

6. Output

Provided by policy. Essential to deliver any document up the stack, events/alerts.

7. Fleet

Provided by policy. Essential to fetch policy update (user artifacts), and upload response of some Response Actions.

**Comment by bmorelli25** [comment]
Related issues:

* https://github.com/elastic/security-docs/issues/4652
* https://github.com/elastic/security-docs/issues/2932

**Comment by natasha-moore-elastic** [comment]
Hi @intxgo, @roxana-gheorghe, I'm not really clear on whether we need any doc updates here or not. If we do, could you clarify what we need to update and on which page? Thanks!

**Comment by intxgo** [comment]
I also don't think we need to explicitly document that, as these are instructions explicitly for being offline.

What was pointed above is that we don't document that:

1. Offline endpoint won't participate with staged artifacts rollout - as the customer is going to manage artifacts themselves.

2. Offline endpoint won't use reputation services aiming to quickly suppress FP (false positives), point 3 online triage of malware alerts, point 4 behavioral reputation rules online triage

It seems the question stems from customers analyzing internet requests made by Endpoint, but it's not clear to me what's the purpose. These additional capabilities are not meant to be replicated by customers in their offline environments. These are services run by Elastic Security (with no static content to copy from), propped by our security research. Customer won't be able to replicate them. They could only just proxy the web requests to our services, but then I see no point in that, as they won't be offline anymore.
